src/UiModes.ts

```diff
@@ -0,0 +1,321 @@
+                UiModes[3].getSettings().push(new SettingData(setting.getKey(),setting.getComponentDefaultVal(),setting.getPerpetual()))
+                UiModes[4].getSettings().push(new SettingData(setting.getKey(),setting.getExpertDefaultVal(),setting.getPerpetual()))
+            })
+        })
+        UiModeSystem.loadFromLocalStorage()
+        const uiModeName = localStorage.getItem('activeUiMode')
+        if(uiModeName === null){
+            UiModeSystem.setActiveUiMode(UiModes[2])
+        }else{
+            UiModeSystem.setActiveUiModeByName(uiModeName)
+        }
+    }
+
+    static saveToLocalStorage = () : void => {
+        if(this.localStorageUpdateCooldown===false){
+            this.localStorageUpdateCooldown = true;
+
+            setTimeout(function () {
+                const uiModesObj : any[] = []
+                UiModeSystem.getUiModes().forEach(function(uiMode:UiMode){
+                    const uiModeObj = {
+                        name : uiMode.getName(),
+                        description : uiMode.getDescription(),
+                        settingValues : <any[]>[]
+                    }
+                    uiMode.getSettings().forEach(function(setting:SettingData){
+                        const settingObj = {
+                            key : setting.getKey(),
+                            value : setting.getValue()
+                        }
+                        uiModeObj.settingValues.push(settingObj)
+                    })
+                    uiModesObj.push(uiModeObj)
+                })
+
+                localStorage.setItem('UiModes', JSON.stringify(uiModesObj));
+                localStorage.setItem('activeUiMode', UiModeSystem.getActiveUiMode().getName());
+                UiModeSystem.localStorageUpdateCooldown = false;
+            }, 1000)
+        }else{
+            return
+        }
+    }
+
+    static loadFromLocalStorage = () : void => {
+        const uiModesObj : any[] = JSON.parse(localStorage.getItem('UiModes'))
+
+        if(uiModesObj === null){
+            return
+        }
+
+        uiModesObj.forEach(function(uiModeObj){
+            let destUiMode : UiMode = UiModeSystem.getUiModeByName(uiModeObj.name)
+            if(destUiMode===null){
+                const settings : SettingData[] = []
+                
+                Setting.getSettings().forEach(function(settingsGroup){
+                    settingsGroup.getSettings().forEach(function(setting){
+                        settings.push(new SettingData(setting.getKey(),setting.getGraphDefaultVal(),setting.getPerpetual()))
+                    })
+                })
+
+                destUiMode = new UiMode(uiModeObj.name, uiModeObj.description, settings, false)
+            }
+            uiModeObj.settingValues.forEach(function(settingObj:any){
+                destUiMode.setSettingByKey(settingObj.key,settingObj.value)
+            } )
+
+        })
+    }
+
+    static updateSettingsArray = () : void => {
```

Comment:
I think this function is not updating the settings array as the name would suggest.

It seems to be overwriting the global settings values with values from the current active ui mode.

I think this should happen automatically when the UIMode is changed. I can't think of another time when it will run.

Maybe we should just move this loop to the place where the UI mode is changed. Or, if you want to keep it as a separate function, move it close to that function and make it a private helper function